src/depotchest.cpp

```diff
@@ -43,23 +43,25 @@ ReturnValue DepotChest::queryAdd(int32_t index, const Thing& thing, uint32_t cou
 	return Container::queryAdd(index, thing, count, flags, actor);
 }
 
-void DepotChest::postAddNotification(Thing* thing, const Cylinder* oldParent, int32_t index, cylinderlink_t)
+void DepotChest::postAddNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> oldParent,
+                                     int32_t index, cylinderlink_t)
 {
-	Cylinder* parent = getParent();
+	auto parent = getParent();
 	if (parent) {
 		parent->postAddNotification(thing, oldParent, index, LINK_PARENT);
 	}
 }
 
-void DepotChest::postRemoveNotification(Thing* thing, const Cylinder* newParent, int32_t index, cylinderlink_t)
+void DepotChest::postRemoveNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> newParent,
+                                        int32_t index, cylinderlink_t)
 {
-	Cylinder* parent = getParent();
+	auto parent = getParent();
```

Comment:
Could this be `const auto`
```suggestion
	const auto parent = getParent();
```